src/main/java/org/openstreetmap/atlas/generator/sharding/PbfVerifier.java

```diff
@@ -0,0 +1,168 @@
+package org.openstreetmap.atlas.generator.sharding;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.stream.Collectors;
+
+import org.openstreetmap.atlas.exception.CoreException;
+import org.openstreetmap.atlas.generator.tools.filesystem.FileSystemHelper;
+import org.openstreetmap.atlas.geography.MultiPolygon;
+import org.openstreetmap.atlas.geography.Polygon;
+import org.openstreetmap.atlas.geography.Rectangle;
+import org.openstreetmap.atlas.geography.atlas.pbf.AtlasLoadingOption;
+import org.openstreetmap.atlas.geography.atlas.raw.creation.RawAtlasGenerator;
+import org.openstreetmap.atlas.streaming.resource.File;
+import org.openstreetmap.atlas.streaming.resource.Resource;
+import org.openstreetmap.atlas.utilities.collections.StringList;
+import org.openstreetmap.atlas.utilities.maps.MultiMap;
+import org.openstreetmap.atlas.utilities.runtime.Command;
+import org.openstreetmap.atlas.utilities.runtime.CommandMap;
+import org.openstreetmap.atlas.utilities.time.Time;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Command that verifies all expected PBF files exist and can be converted to atlas.
+ *
+ * @author jamesgage
+ */
+public class PbfVerifier extends Command
+{
+    private static final Switch<File> SLIPPY_TILE_FILE = new Switch<>("slippyTileFile",
+            "The slippy tile file that lists all pbf files that should have been generated.",
+            File::new, Optionality.REQUIRED);
+    private static final Switch<String> PBF_PATH = new Switch<>("pbfPath",
+            "The path to the pbf root directory", value -> value, Optionality.REQUIRED);
+    private static final int LOGGING_RATE = 1000;
+    private static final Logger logger = LoggerFactory.getLogger(PbfVerifier.class);
+
+    public static MultiPolygon forPolygon(final Polygon polygon)
+    {
+        final MultiMap<Polygon, Polygon> multiMap = new MultiMap<>();
+        multiMap.put(polygon, new ArrayList<>());
+        return new MultiPolygon(multiMap);
+    }
+
+    public static void main(final String[] args)
+    {
+        new PbfVerifier().run(args);
+    }
+
+    public static HashMap<String, Rectangle> parseSlippyTileFile(final Resource slippyTileFile)
+    {
+        final Integer size = Integer.parseInt(slippyTileFile.firstLine());
+        final HashMap<String, Rectangle> shardToBounds = new HashMap<>(size);
+        for (final String line : slippyTileFile.lines())
+        {
+            final StringList splitLine = StringList.split(line, "|");
+            // the form of the slippyTile file is "pbfName.pbf|POLYGON( BOUNDS )"
+            if (splitLine.size() == 2)
```

Comment:
If the split isn't 2, should we log that the input is corrupt?

Reply:
The first line of the slippyTile is just the number of pbfs within the file, so the first line would fail that check. I'll add some error handling though.